packages/esm-patient-chart-app/src/ui-components/action-menu.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useCallback } from "react";
+import {
+  Edit20,
+  Notification20,
+  ShoppingBag20,
+  Document20,
+} from "@carbon/icons-react";
+import Button from "carbon-components-react/es/components/Button";
+import { HeaderPanel } from "carbon-components-react";
+
+import styles from "./action-menu.scss";
+import {
+  ActionMenuHookState,
+  ActionMenuTypes,
+} from "../hooks/useActionMenuState";
+import { isDesktop } from "../utils";
+import { useLayoutType } from "@openmrs/esm-framework";
+
+interface ActionMenuInterface {
+  config: ActionMenuHookState;
+}
+
+export const ActionMenu: React.FC<ActionMenuInterface> = ({ config }) => {
+  const layout = useLayoutType();
+  const createToggle = (type: ActionMenuTypes) => () =>
+    config.setActionMenuState(
+      config.state.type === type ? config.defaultState : { type }
+    );
+  const onNotificationsClick = useCallback(
+    createToggle(ActionMenuTypes.NOTIFICATIONS),
+    [config.state.type]
+  );
+  const onCartClick = useCallback(createToggle(ActionMenuTypes.CART), [
+    config.state.type,
+  ]);
+  const onDocumentsClick = useCallback(
+    createToggle(ActionMenuTypes.DOCUMENTS),
+    [config.state.type]
+  );
+
+  const menu = isDesktop(layout) ? (
+    <aside className={styles.rightSideNav}>
+      <Button
```

Comment:
Should these buttons be hard coded in here?

Some of them don't seem to be related to patient chart at all. Please offer a proper extension slot.

Reply:
hmmm gotcha - time to finally learn extensions... will get that to you soon. gonna have to think through creating an extension slot for the drawer content as well, i think